src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -240,16 +331,22 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
 
+        /// <summary>
+        /// Disposes of resources.
+        /// </summary>
+        /// <param name="disposing">Indicates if managed code is being disposed.</param>
         private void Dispose(bool disposing)
         {
-            if (disposing)
+            if (!this.isDisposed && disposing)
             {
+                this.isDisposed = true;
                 Interlocked.CompareExchange(ref active, null, this);
 
                 ITelemetryChannel telemetryChannel = this.TelemetryChannel;
                 if (telemetryChannel != null)
                 {
                     telemetryChannel.Dispose();
+                    telemetryChannel = null;
```

Comment:
I was trying to be friendly to the garbage collector, it may not really be necessary these days but it was a common pattern to null references after disposing.